server/src/main/java/org/elasticsearch/action/support/IndicesOptions.java

```diff
@@ -286,20 +300,28 @@ public static Builder builder(WildcardOptions wildcardOptions) {
     }
 
     /**
-     * These options apply on all indices that have been selected by the other Options. It can either filter the response or
-     * define what type of indices or aliases are not allowed which will result in an error response.
+     * The "gatekeeper" options apply on all indices that have been selected by the other Options. It contains two type of flags:
+     * - The "allow*" flags, which purpose is to enable actions to define certain conditions that need to apply on the concrete indices
+     * they accept. For example, single-index actions will set allowAliasToMultipleIndices to false, while search will not accept a
+     * closed index etc. These options are not configurable by the end-user.
+     * - The ignoreThrottled flag, which is a depricared flag that will filter out frozen indices.
      * @param allowAliasToMultipleIndices, allow aliases to multiple indices, true by default.
      * @param allowClosedIndices, allow closed indices, true by default.
-     * @param ignoreThrottled, filters out throttled (aka frozen indices), defaults to true.
+     * @param allowFailureIndices, allow failure indices in the response, true by default
+     * @param ignoreThrottled, filters out throttled (aka frozen indices), defaults to true. This is deprecated and the only one
+     *                         that only filters and never throws an error.
      */
-    public record GeneralOptions(boolean allowAliasToMultipleIndices, boolean allowClosedIndices, @Deprecated boolean ignoreThrottled)
-        implements
-            ToXContentFragment {
+    public record GatekeeperOptions(
```

Comment:
I can also apply this renaming in a different PR if it's too much.

Reply:
Nah, I wouldn't worry about it. I think it's fine here!